### PR TITLE
Do not serve assets from CA's backend

### PR DESCRIPTION
**Pull request by emmachughes**
The `/content/assets/...` endpoint in Content Author is used to serve files from the storage. However, it turns out there's a much better solution for serving files: it's called "using a web server".

Serving files from CA's PHP backend itself has some problems:

- Large files can cause OoM due to a bug with streamed responses in Laravel. Apparently the workaround isn't compatible with S3.
- It ties up a php-fpm worker while the file is being served.
- It forgoes the benefits of using a CDN, such as caching.

To solve this, we set the `url` option in Laravel's filesystems to the env variable already used for CDN prefixes, then use Laravel's method for generating a 'cloud' URL.

Due to the `/content/assets/` path working its way into databases and stuff, we have to keep supporting it perpetually. For the local environment, we alias it in nginx. For the hosted environments, we'll have to configure the kubernetes ingresses to redirect.

While the `getAssetUrl` method had a `$private` parameter that's removed with this PR, it was practically worthless, as there was no separation between public and private files--every file in the S3 bucket was publicly available through the endpoint.

### CODE REVIEW

**Review by chrieinv — APPROVED** [review]

**Comment by emmachughes** [comment]
I'll wait until #2291 is merged before merging this.

**Pull request merged**
